kairon/shared/concurrency/actors/analytics_runner.py

```diff
@@ -81,14 +83,24 @@ def execute(self, source_code: Text, predefined_objects: Optional[Dict] = None,
             if process.returncode != 0:
                 raise AppException(f"Subprocess error: {stdout.strip()}")
 
+            triggers = action.get("triggers")
+            if triggers and triggers[0].get("conditions") == TriggerCondition.success.value:
+                self.trigger_email(action, bot)
+
             result = json.loads(stdout)
             return self.__cleanup(result)
 
         except Exception as e:
             msg = stdout.strip() if 'stdout' in locals() and stdout else str(e)
             logger.exception(msg)
-            raise AppException(f"Execution error: {msg}") from e
+            triggers = action.get("triggers")
+            if triggers and triggers[0].get("conditions") == TriggerCondition.failure.value:
```

Comment:
iterate over list instead of passing index and check action type condition as well